forem/forem
# Conditionally rendering RSS feed settings

## What type of PR is this? (check all applicable)


## Description

With this commit, we're only allowing users who can create articles to
enter RSS feed information for fetching of articles.

Note: I chose not to indent body of the if conditional to ease the code
review.  I also chose to place the if statement in this partial instead
of in the [app/views/users/_extensions.html.erb][1] file; this helps
contain the logic around the policy.

[1]:https://github.com/forem/forem/blob/b87fd77992fc8c9f2c53da86498e44a052da2e97/app/views/users/_extensions.html.erb#L3

## Related Tickets & Documents

For sleuthing this mirrors the approach of forem/forem#16735

Closes forem/forem#16788
Related to forem/forem#16766, forem/forem#16732, and forem/forem#16763

## QA Instructions, Screenshots, Recordings

The AritclePolicy tests the permutations of users who can update these settings.  So I'm going to rely on "our existing test suite" already exercises this area of the code.

But, if you were one to want to test this in the UI, you'd want to create a non-administrative user, enable the `:limit_post_creation_to_admins` feature flag, then go to `/settings/extensions` and notice the missing "Publishing to RSS" section.

### UI accessibility concerns?

None.

## Added/updated tests?

- [x] No, and this is why: See QA instructinos

## [Forem core team only] How will this change be communicated?
